pipenv/patched/notpip/_internal/index.py

```diff
@@ -766,7 +766,7 @@ def egg_info_matches(
         return full_match[full_match.index('-'):]
     name = match.group(0).lower()
     # To match the "safe" name that pkg_resources creates:
-    name = name.replace('_', '-')
+    name = name.replace('_', '-').replace('.', '-')
```

Comment:
^_^ I am actually tempted to do this, but I have a preference for not patching pip whenever possible.

I'm likely going to hold off on accepting this for the next release to see about getting feedback from the pip maintainers when I have a bit more clarity on all of the places we are currently working around this problem, which I think is pretty numerous by now...

Reply:
It's up to you if you accept it, and I understand if you do not, but this is currently breaking our (am I am sure others) workflow entirely. I can't see a workaround apart from downgrading

Reply:
@orf, apart from the issues mentioned and the fixes for those specifically (which are in master), can you describe what is continuing to cause you problems? I.e. with the current master branch, how do I see a problem?

Reply:
I will get back to you here tomorrow morning, after I try installing the specific pipfile I have with the current master

Reply:
ah, I see the issue in question.  This should be resolved by #3007

Reply:
Awesome! Thanks for your continued work on pipenv, and I'm sorry for the volume of issues!

I will confirm tomorrow that this is fixing it, or though I cannot see any code that would after my brief lookm